app/controllers/AdditionalPaymentAmountController.scala

```diff
@@ -55,7 +55,9 @@ class AdditionalPaymentAmountController @Inject()(
     getRequiredAnswerOrRedirectV(AdditionalPaymentPeriodsPage) { additionalPaymentPeriods =>
       withValidAdditionalPaymentDate(additionalPaymentPeriods, idx) { paymentDate =>
         val preparedForm = request.userAnswers.getV(AdditionalPaymentAmountPage, Some(idx)) match {
-          case Invalid(e)   => form
+          case Invalid(e) =>
+            UserAnswers.logWarnings(e)(logger)
```

Comment:
instead of logging every where inside controllers, is it possible to just log once inside `getV` ?  
this way it would be like several logs in controllers vs single log inside `getV`

Reply:
@sureshhmrc Of course, it's just that we have several methods, and I thought we may want to add extra logging. The problem with doing in `getV` is that you wouldn't get location pass through. E.g you'd always get the same logger.

Reply:
@sureshhmrc The only other approach would be to pass through a logger implicitly from the callsite to `getV` methods. Simply logging in `getV` will not be as helpful to us as we want I believe. That's why I did it this way.

Reply:
@sureshhmrc Also, it's not always appropriate to log in `getV`. In many instances, we just have warnings not errors. How could you differentiate without the business logic context? How do you know you only need to log at the end of 3 or 4 `getV` calls for exampe?

Reply:
the `logWarnings` can simply be called from `getV` with implicit logger in place ? so no need to controller changes (except implicit logger), 
but anyway its fine with me as its already approved, feel free to merge, thanks

Reply:
@sureshhmrc How do you know inside `getV` when you need to log a warning or an error?

Reply:
actually why do we have both `warn` and `error` ?  I remember logging at `error` level means that something very serious have happened which needs immediate attention (including platops)
if its `InValid(e)` it should always be a warn (or an error ) to keep it consistant..
i may have missed some business requirements here, so not sure if we need to have both warns and errors

Reply:
@sureshhmrc Worth having a look through the code to understand the discrepancies and the reason why we should clearly differentiate between log levels.

Reply:
https://confluence.tools.tax.service.gov.uk/display/DTRG/Logging+best+practices says 
```Use the appropriate log level
Error - for significant issues that you will cause someone to be woken up at night to fix, such as inability to perform the service's central responsibility. These should very rarely happen. An example is an OutOfMemoryError or IO error.
Warn - for significant issues that will get someone's attention. Examples include unexpected data returned from a dependent service.
Info - general information useful to someone monitoring the service, such as you do after a deploy. Examples include payload.
Debug - detailed information from internal methods used for developers. This must be switched off in Production and Staging.
```

Reply:
sure, will go through the code at some point but based on above guidance, we should not log at error level - and use only use warn for almost every error. thanks
but feel free to merge - we can revisit later